sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -89,6 +89,17 @@ abstract class BinaryExpression extends Expression with trees.BinaryNode[Express
   override def toString: String = s"($left $symbol $right)"
 }
 
+/**
+ *  This class is for expressions that use math functions like `pow`, `max`, `hypot`, and `atan2`,
+ *  which can't be expressed as `$left $symbol $right`. They are expressed as
+ *  `$func($left, $right)`.
+ */
+abstract class BinaryFunctionExpression extends Expression with trees.BinaryNode[Expression] {
```

Comment:
is the only difference toString? In that case can't we just override toString in BinaryMathExpression?